pkg/component/controller/controllersleasecounter.go

```diff
@@ -56,7 +57,8 @@ func (l *K0sControllersLeaseCounter) Run(ctx context.Context) error {
 	}
 
 	// hostname used to make the lease names be clear to which controller they belong to
-	holderIdentity, err := os.Hostname()
+	// follow kubelet convention for naming so we e.g. use lowercase hostname etc.
+	holderIdentity, err := nodeutil.GetHostname("")
```

Comment:
What about the "override-hostname" stuff? Are we ignoring that?

Reply:
Yes, all this stuff is happening on controllers only and "override-hostname" is for kubelet.